packages/commons/src/conditionals/validate.ts

```diff
@@ -126,9 +130,20 @@ export function validate(schema: JSONSchema, data: ConditionalParameters) {
       Object.entries(data.$form).map(([key, value]) => {
         const maybeAgeValue = AgeValue.safeParse(value)
         if (maybeAgeValue.success) {
+          const age = maybeAgeValue.data.age
+          const maybeAsOfDate = DateValue.safeParse(
+            data.$form[maybeAgeValue.data.asOfDateRef]
+          )
+
           return [
             key,
-            { age: maybeAgeValue.data.age, dob: ageToDate(maybeAgeValue.data) }
+            {
+              age,
+              dob: ageToDate(
+                age,
+                maybeAsOfDate.success ? maybeAsOfDate.data : data.$now
```

Comment:
Could this cause confusion? I mean when the asOfDate is not inputted yet, the validations would work differently right?